docs/reference/connector/apis/update-connector-filtering-api.asciidoc

```diff
@@ -0,0 +1,186 @@
+[[update-connector-filtering-api]]
+=== Update connector filtering API
+
+preview::[]
+
+++++
+<titleabbrev>Update connector filtering</titleabbrev>
+++++
+
+Updates the `filtering` configuration of a connector.
+
+[[update-connector-filtering-api-request]]
+==== {api-request-title}
+
+`PUT _connector/<connector_id>/_filtering`
+
+[[update-connector-filtering-api-prereq]]
+==== {api-prereq-title}
+
+* To sync data using connectors, it's essential to have the Elastic connectors service running.
+* The `connector_id` parameter should reference an existing connector.
+
+[[update-connector-filtering-api-path-params]]
+==== {api-path-parms-title}
+
+`<connector_id>`::
+(Required, string)
+
+[role="child_attributes"]
+[[update-connector-filtering-api-request-body]]
+==== {api-request-body-title}
+
+`filtering`::
+(Required, array) The filtering configuration for the connector. This configuration determines the set of rules applied for filtering data during syncs.
+
+Each entry in the `filtering` array represents a set of filtering rules for a specific data domain and includes the following attributes:
+
+- `domain` (Required, string) +
+Specifies the data domain to which these filtering rules apply.
+
+- `active` (Required, object) +
+Contains the set of rules that are actively used for sync jobs. The `active` object includes:
+
+  * `rules` (Required, array of objects) +
+  An array of individual filtering rule objects, each with the following sub-attributes:
+    ** `id` (Required, string) +
+    A unique identifier for the rule.
+    ** `policy` (Required, string) +
+    Specifies the policy, such as "include" or "exclude".
+    ** `field` (Required, string) +
+    The field in the document to which this rule applies.
+    ** `rule` (Required, string) +
+    The type of rule, such as "regex", "starts_with", "ends_with", "contains", "equals", "<", ">", etc.
+    ** `value` (Required, string) +
+    The value to be used in conjunction with the rule for matching the contents of the document's field.
+    ** `order` (Required, number) +
+    The order in which the rules are applied. The first rule to match has its policy applied.
+    ** `created_at` (Optional, datetime) +
+    The timestamp when the rule was added.
+    ** `updated_at` (Optional, datetime) +
+    The timestamp when the rule was last edited.
+
+  * `advanced_snippet` (Optional, object) +
+  Used for advanced filtering at query time, with the following sub-attributes:
```

Comment:
```suggestion
  Used for {enterprise-search-ref}/sync-rules.html#sync-rules-advanced[advanced filtering] at query time, with the following sub-attributes:
```

Link to https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-advanced?